util/types/datum.go

```diff
@@ -1447,3 +1448,39 @@ func EqualDatums(a []Datum, b []Datum) (bool, error) {
 	}
 	return true, nil
 }
+
+// DatumEquals checker.
+type datumEqualsChecker struct {
+	*check.CheckerInfo
+}
+
+// DatumEquals checker verifies that the obtained value is equal to
+// the expected value.
+// For example:
+//     c.Assert(value, DatumEquals, NewDatum(42))
+var DatumEquals check.Checker = &datumEqualsChecker{
```

Comment:
We should avoid import test packages in non-test code. because it generates a lot of test flags in the executable file.